tests/check_check_tags.c

```diff
@@ -0,0 +1,621 @@
+/*
+ * Check: a unit test framework for C
+ * Copyright (C) 2001, 2002 Arien Malec
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+#include "../lib/libcompat.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <check.h>
+#include <assert.h>
+#include "check_check.h"
+
+enum {
+    RED,
+    BLUE,
+    PURPLE,
+    YELLOW,
+    BLACK,
+    MAX_TESTS
+} test_ids;
+
+unsigned int test_executed[MAX_TESTS];
+
+static void reset_executed(void)
+{
+    unsigned int i;
+
+    for (i = 0; i < MAX_TESTS; i++)
+    {
+	test_executed[i] = 0;
+    }
+}
+
+START_TEST(red_test1)
+{
+    test_executed[RED] = 1;
```

Comment:
I think what I'm doing here is the same as is used by check_check_selective - I agree it depends on forking being disabled but in both selective_setup() and tag_test_setup() the fork_status is explicitly set to CK_NOFORK.

My question is whether limiting the testing to NOFORK mode matters ? I feel I have enough of an overview to judge that so I'll be guided by you. 

In terms of the code it would be easy enough for me to go back to the assert driven scheme but in my view there is a slight readability and code size cost to that approach - and I should probably do some tweaks to check_check_main/master to properly share the pieces of code I need to help handle the logging.